addons/website/views/website_templates.xml

```diff
@@ -111,13 +111,16 @@
             </t>
         </t>
         <!-- `alternate`/`canonical` mainly useful to crawlers/bots/SEO tools, which test the website as public user -->
-        <t t-if="request and request.is_frontend_multilang and website and website.is_public_user()">
+        <t t-if="request and request.is_frontend_multilang">
             <t t-set="alternate_languages" t-value="website._get_alternate_languages(canonical_params=canonical_params)"/>
-            <t t-foreach="alternate_languages" t-as="lg">
-        <link rel="alternate" t-att-hreflang="lg['hreflang']" t-att-href="lg['href']"/>
-            </t>
+            <t t-foreach="alternate_languages" t-as="lg" t-if="len(alternate_languages) > 1">
+        <link rel="alternate" t-att-hreflang="lg['url_code']" t-att-href="lg['href']"/>
+        <link rel="alternate" hreflang="x-default" t-att-href="lg['href']" t-if="lg['id'] == website._get_cached('default_lang_id')"/>
+        <link rel="canonical" t-att-href="lg['href']" t-if="lg['code']==lang"/>
+                </t><t t-else="">
+        <link rel="canonical" t-att-href="alternate_languages[0]['href']"/>
+                </t>
```

Comment:
Same indentation as the openin `t-if`?
```suggestion
            </t><t t-else="">
        <link rel="canonical" t-att-href="alternate_languages[0]['href']"/>
            </t>
```

But I actually don't get why we need to do differently if we have 2 langs or 3+ langs

Reply:
The difference is between 1 and 2+; when we have more than two, we have to pick the right one. (the indentation was confusing, just fixed it)